CMakeLists.txt

```diff
@@ -465,12 +465,19 @@ if(EXISTS ${CMAKE_BINARY_DIR}/zephyr_modules.txt)
     # lazy regexes (it supports greedy only).
     string(REGEX REPLACE "\"(.*)\":\".*\"" "\\1" module_name ${module})
     string(REGEX REPLACE "\".*\":\"(.*)\"" "\\1" module_path ${module})
+
+    # Paths used in zephyr_modules.txt might use '\' as path separator.
+    # This leads to errors in certain cases where ${ZEPHYR_CURRENT_MODULE_DIR} is used.
+    # As zephyr module requires a CMakeLists.txt to be present
+    # the find_path command can be used to convert to CMake style path.
+    find_path(cmake_module_path "CMakeLists.txt" PATH ${module_path})
+    set(ZEPHYR_CURRENT_MODULE_DIR ${cmake_module_path})
+
```

Comment:
I wonder how this works, find_path should be caching it's result and therefore this should be broken, but evidently it is not.

Reply:
Has it been considered to write zephyr_modules.txt with '/' as path separator? Then the next user of zephyr_modules.txt won't have to solve the same problem.

Reply:
> find_path avoids issues later if multiple path seperators are
> used, such as 'path/////to///somewhere when striping.

This is a valid point. But again, if we could avoid generating double-slashes in the first place then other tooling that integrates with zephyr_modules.txt will be protected as well.

Reply:
> > find_path avoids issues later if multiple path seperators are
> > used, such as 'path/////to///somewhere when striping.
> 
> This is a valid point. But again, if we could avoid generating double-slashes in the first place then other tooling that integrates with zephyr_modules.txt will be protected as well.

Very good point.
PR changed to ensure path is posix style after zephyr_module.py